Shopify/hydrogen
# (Unstable) server-side network request debug route

### WHAT is this pull request doing?

(Unstable) A new virtual route that logs the network requests happening at the server side.

<img width="1085" alt="Screenshot 2023-09-06 at 1 43 54 PM" src="https://github.com/Shopify/hydrogen/assets/2319002/1bd9b979-35ff-44f8-ac94-81439aeec989">


#### Features

* Log the main request and the sub requests
* Main request
  * URL format: `<cache status> <purpose?> <url>`
  * (Green bar) Time spent on server
  * (Blue bar) Time spent on streaming
* Sub request
  * URL format: `<cache status> <purpose?> <query name>`
* Summary bar
* Option to show PUT requests

#### Known issues

* Sometimes, request/sub request don't get logged - Rerun `npm run dev` will fix this

#### How to test

After running `npm run dev` in a project, you should see

<img width="590" alt="Screenshot 2023-08-31 at 10 42 57 AM" src="https://github.com/Shopify/hydrogen/assets/2319002/f0edb4c7-3fe7-4744-ba76-772e206dfa41">

Open http://localhost:3000/debug-network

<img width="1083" alt="Screenshot 2023-09-06 at 1 46 33 PM" src="https://github.com/Shopify/hydrogen/assets/2319002/568cad7b-70e9-457f-bec2-4e59d72daa54">

Clicking on the link will open http://localhost:3000 in another tab.

Navigate around the app and you should see server-side network requests being logged in the network flame graph

#### Post-merge steps



#### Checklist
